R/merge.R

```diff
@@ -52,32 +52,41 @@
 
 #' @importFrom S4Vectors SimpleList
 #' @importFrom scuttle sumCountsAcrossFeatures
-.merge_rows <- function(x, f, archetype = 1L, 
+.merge_rows <- function(x, f, archetype = 1L,
+                        na.rm = FALSE,
                         average = FALSE,
                         BPPARAM = SerialParam(),
                         check.assays = TRUE,
                         ...){
     # input check
+    if(!.is_a_bool(na.rm)){
+        stop("'na.rm' must be TRUE or FALSE.", call. = FALSE)
+    }
     if( !.is_a_bool(average) ){
         stop("'average' must be TRUE or FALSE.", call. = FALSE)
     }
     if( !.is_a_bool(check.assays) ){
         stop("'check.assays' must be TRUE or FALSE.", call. = FALSE)
     }
-    #
+
+    # This is done otherwise we lose NA values
+    if ( na.rm ) {
+        f[is.na(f)] <- "NA"
+    }
     if(is.character(f) && length(f)==1 && f %in% colnames(rowData(x))){
         f <- factor(as.character(rowData(x)[, f]))
     }
     else if(is.character(f) && length(f)==1 && f %in% colnames(colData(x))){
+        f[is.na(f)] <- "NA"
         f <- factor(as.character(colData(x)[, f]))
-    } else 
+    } else
     {
-        f <- .norm_f(nrow(x), f)  
+        f <- .norm_f(nrow(x), f)
     }
```

Comment:
1. Remove na.rm from .merge_rows and .merge_cols
2. Modify these if sentences that check whether f is name from col/rowData or vector. Something like this:
```

if( .is_a_string(f) && f %in% colnames(rowData(x)) ){
    f <- rowData(x)[[ f ]]
}
f <- .norm_f(nrow(x), f)
```
3. Add na.rm to .norm_f function

Reply:
ACK